common/level.go

```diff
@@ -5,44 +5,146 @@ import (
 	"engo.io/gl"
 )
 
+// Level is a parsed TMX level containing all layers and default Tiled attributes
 type Level struct {
-	width      int
-	height     int
-	TileWidth  int
+	// Orientation is the parsed level orientation from the TMX XML, like orthogonal
+	Orientation string
+	// RenderOrder is the in Tiled specified TileMap render order, like right-down
+	RenderOrder string
+	width       int
+	height      int
+	// TileWidth defines the width of each tile in the level
+	TileWidth int
+	// TileHeight defines the height of each tile in the level
 	TileHeight int
-	Tiles      []*tile
+	// NextObjectId is the next free Object ID defined by Tiled
+	NextObjectId int
+	// TileLayers contains all TileLayer of the level
+	TileLayers []*TileLayer
+	// ImageLayers contains all ImageLayer of the level
+	ImageLayers []*ImageLayer
+	// ObjectLayers contains all ObjectLayer of the level
+	ObjectLayers []*ObjectLayer
+}
+
+// TileLayer contains a list of its tiles plus all default Tiled attributes
```

Comment:
I don't like how this is worded, I'd mention something like `represents a tiled map layer in a tmx file` or something like that